packages/salesforcedx-vscode-core/src/conflict/persistentStorageService.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { FileProperties } from '@salesforce/source-deploy-retrieve/lib/src/client/types';
+import {
+  ExtensionContext,
+  Memento
+} from 'vscode';
+
+interface ConflictFileProperties {
+  lastModifiedDate: string;
+}
+
+export class PersistentStorageService {
+  private storage: Memento;
+  private static instance?: PersistentStorageService;
+
+  private constructor(context: ExtensionContext) {
+    this.storage = context.workspaceState;
```

Comment:
it looks like we switched over to using the `workspaceState` storage but from my local testing, I'm not seeing this storage persist data between sessions. are there any drawbacks to using the `globalState` storage?

Reply:
My thought was just that using the smaller scope of workspaceState would be preferable if it did in fact work. I saw the data persist for simple cases but did not thoroughly test it, so I have switched back to global in the latest commit.